app/assets/javascripts/components/user_photo_edit.es6.jsx

```diff
@@ -0,0 +1,201 @@
+          photo: this.state.render_photo,
+          title: this.state.title,
+          description: this.state.description,
+          id: this.state.id,
+        },
+        user_id: this.props.user.id
+      }
+      APIRequester.put(`/images/${this.state.id}`, imageFields, this._successfulSave);
+    } else {
+      toastr.options.positionClass = 'toast-bottom-right';
+      toastr.error("Please select an image.");
+    }
+  }
+
+  _successfulSave() {
+    window.location = location.pathname;
+    this.setState({ show_form: false });
+    this.setState({ showModal: false });
+  }
+
+  renderPhotoSelection() {
+    return (
+      <div>
+        <button className='btn btn-sm btn-action btn-action pull-left' onClick={this._openModal}>Edit Photo</button>
+        <Modal className="modal" show={this.state.showModal}>
+          <Modal.Header>
+            <Modal.Title>Select a Photo to Edit</Modal.Title>
+          </Modal.Header>
+            <Modal.Body>
+              <div className="input-field">
+                Select the image you would like to edit.
+              </div>
+              { this._renderImages() }
+            </Modal.Body>
+            <Modal.Footer>
+              <button className="btn btn-outline" type="button" onClick={this._cancelEdit}>Close</button>
+              <button className="btn btn-blue modal-btn" type="button" onClick={this._enableEdit}>Select</button>
+            </Modal.Footer>
+        </Modal>
+      </div>
+    )
+  }
+
+  renderEdit() {
+    return (
+      <div>
+        <button className='btn btn-sm btn-action btn-action pull-left' onClick={this._openModal}>Edit Photo</button>
+        <Modal className="modal" show={this.state.showModal}>
+          <Modal.Header>
+            <Modal.Title>Edit Photo</Modal.Title>
+          </Modal.Header>
+            <Modal.Body>
+              <div className="input-field">
+              <div className="imgPreview">
+                <img className="profile-preview" src={this.state.render_photo} />
+              </div>
+              <label className="file-label" htmlFor="file-input">Choose a Photo</label>
+              <input className="inputfile" id="file-input" type="file" name="file" onChange={this._handleFileChange} />
+              </div>
+              <div className="input-field">
+                <label htmlFor="title-input">Title</label>
+                <input id="title-input" type="text" name="title" defaultValue={this.state.title} onChange={this._handleTextChange} />
+              </div>
+              <div className="input-field">
+                <label htmlFor="description-input">Description</label>
+                <input id="description-input" type="text" name="description" defaultValue={this.state.description} onChange={this._handleTextChange} />
+              </div>
+            </Modal.Body>
+            <Modal.Footer>
+              <button className="btn btn-outline" type="button" onClick={this._cancelEdit}>Close</button>
```

Comment:
nit: add "modal-btn" to the className for "close" and "back" buttons to add horizontal space between the buttons

![screen shot 2017-04-13 at 12 59 41 pm](https://cloud.githubusercontent.com/assets/10657390/25022061/545ffd74-2049-11e7-8623-015278fcf33f.png)